### PR TITLE
Jun 2 Fixes

### PR DESCRIPTION
**Pull request by tsortwehttam**
OK to merge.

Note: This is branched from `jun1-fixes`.

Summary of changes:

- Various Asana tasks filed by Taylor on Friday. See individual commit messages for deets.

### CODE REVIEW

**Comment by stristr** [comment]
(cool, I figured out how to trick Github into actually showing the commits that would be new to `master`)

**Comment by tsortwehttam** [comment]
BUILDME!

File: packages/haiku-glass/src/react/Glass.js [hunk 1]
```diff
@@ -660,6 +660,31 @@ export class Glass extends React.Component {
           )
           break
 
+        case 'instantiate-component':
+          const component = this.getActiveComponent()
```
Comment (roperzh) [inline packages/haiku-glass/src/react/Glass.js:664]: feels way nicer to have this abstracted here :)

**Review by roperzh — APPROVED** [review]
LGTM

File: packages/haiku-serialization/src/bll/Element.js [hunk 2]
```diff
@@ -1711,7 +1711,7 @@ class Element extends BaseModel {
           // is clipped.
           'style.overflow': 'visible',
           [HAIKU_SOURCE_ATTRIBUTE]: `${svgElement.attributes[HAIKU_SOURCE_ATTRIBUTE]}#${descendantHaikuElement.id}`,
-          [HAIKU_TITLE_ATTRIBUTE]: descendantHaikuElement.title || descendantHaikuElement.id
+          [HAIKU_TITLE_ATTRIBUTE]: descendantHaikuElement[HAIKU_TITLE_ATTRIBUTE] || descendantHaikuElement.title || descendantHaikuElement.id
```
Comment (stristr) [inline packages/haiku-serialization/src/bll/Element.js:1714]: @matthewtoast—isn't this (`descendantHaikuElement[HAIKU_TITLE_ATTRIBUTE]`) superfluous and not where the attribute actually goes?

Comment (tsortwehttam) [reply packages/haiku-serialization/src/bll/Element.js:1714]: Nope, this is correct: https://github.com/HaikuTeam/mono/pull/514/commits/f3f1ea4b7804507c9dea5356163f9cc5af26e3a2#diff-b80e672eadaf866a5d52c67df4456203R162

Comment (stristr) [reply packages/haiku-serialization/src/bll/Element.js:1714]: Doesn't that line set it on `attributes`, not the `HaikuElement` directly?

Which `.title` is actually just a getter for?

Comment (tsortwehttam) [reply packages/haiku-serialization/src/bll/Element.js:1714]: Ahh, I totally forgot about that getter even existing. `descendantHaikuElement.title` should be sufficient here.

**Pull request merged**